plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/DefaultGlueColumnStatisticsProvider.java

```diff
@@ -247,49 +248,81 @@ public void updateTableColumnStatistics(Table table, Map<String, HiveColumnStati
-                    .addAll(deleteStatsFutures)
-                    .build();
+    // If a whole batch of updates are available, parallelize over Partitions
+    @Override
+    public void updatePartitionStatistics(Set<StatisticsUpdate> statisticsUpdates)
+    {
+        List<CompletableFuture<Void>> operationsFutures = new ArrayList<>();
+        for (StatisticsUpdate update : statisticsUpdates) {
+            operationsFutures.add(runAsync(() -> {
+                Partition partition = update.getPartition();
+                Map<String, HiveColumnStatistics> updatedColumnStatistics = update.getColumnStatistics();
 
-            getFutureValue(allOf(updateOperationsFutures.toArray(CompletableFuture[]::new)));
-        }
-        catch (RuntimeException ex) {
-            throw new TrinoException(HIVE_METASTORE_ERROR, ex);
+                HiveBasicStatistics partitionStats = getHiveBasicStatistics(partition.getParameters());
+                List<ColumnStatistics> columnStats = toGlueColumnStatistics(partition, updatedColumnStatistics, partitionStats.getRowCount()).stream()
+                        .filter(this::isGlueWritable)
+                        .collect(toUnmodifiableList());
+
+                List<List<ColumnStatistics>> columnChunks = Lists.partition(columnStats, GLUE_COLUMN_WRITE_STAT_PAGE_SIZE);
+
+                columnChunks.forEach(columnChunk ->
+                        glueClient.updateColumnStatisticsForPartition(
+                                new UpdateColumnStatisticsForPartitionRequest()
+                                        .withCatalogId(catalogId)
+                                        .withDatabaseName(partition.getDatabaseName())
+                                        .withTableName(partition.getTableName())
+                                        .withPartitionValues(partition.getValues())
+                                        .withColumnStatisticsList(columnChunk)));
+
+                Map<String, HiveColumnStatistics> currentColumnStatistics = this.getPartitionColumnStatisticsIfPresent(partition).orElse(ImmutableMap.of());
+                Set<String> removedStatistics = difference(currentColumnStatistics.keySet(), updatedColumnStatistics.keySet());
+                removedStatistics.forEach(column ->
+                        glueClient.deleteColumnStatisticsForPartition(
+                                new DeleteColumnStatisticsForPartitionRequest()
+                                        .withCatalogId(catalogId)
+                                        .withDatabaseName(partition.getDatabaseName())
+                                        .withTableName(partition.getTableName())
+                                        .withPartitionValues(partition.getValues())
+                                        .withColumnName(column)));
+            }, writeExecutor));
         }
+        getFutureValue(allOf(operationsFutures.toArray(CompletableFuture[]::new)));
```

Comment:
This will propagate failures as RuntimeExceptions - should be proper TrinoExceptions IMO.